src/codec.rs

```diff
@@ -45,13 +50,14 @@ impl Codec {
             &Codec::Null => (),
             &Codec::Deflate => {
                 let mut encoder = Encoder::new(Vec::new());
-                encoder.write(stream)?;
+                encoder.write_all(stream)?;
                 *stream = encoder.finish().into_result()?;
             },
-            #[cfg(feature = "snappy")] &Codec::Snappy => {
+            #[cfg(feature = "snappy")]
+            &Codec::Snappy => {
                 let mut writer = Writer::new(Vec::new());
                 writer.write(stream)?;
```

Comment:
Given the change clippy made L53 of the same file, shouldn't this be `writer.write_all`?

Snap's `Writer` [implements the `io::Write`trait](https://docs.rs/snap/0.2.4/snap/struct.Writer.html#impl-Write), so I suppose that applies here as well?

Maybe clippy does not go inside feature code by default?

Reply:
Nice finding! You are correct, I have to pass to clippy the features the code has to be compiled with. I updated the Makefile and fixed the above issue for snappy as well.